### DISCUSSION TITLE
Response time to process requests in the logs

### DISCUSSION THREAD

**Discussion by Okiharaherbst** [PostgREST/postgrest#2645] [Q&A]
Currently PostgREST outputs these logs:

`[02/Feb/2023:12:27:11 +0100] "POST /rpc/interv_status HTTP/1.1" 200 - "" "python-requests/2.27.1"`

It would be useful to see the response time to serve the request in seconds or milliseconds as that would allow profiling RPC calls for instance.

Is there any way to set this up?

**Comment by steve-chavez** [comment]
How about using Nginx for this? You can configure an `upstream_response_time` to show in the logs.

See:

- https://onecompiler.com/questions/3src8rjkd/how-to-log-application-response-time-with-nginx
- https://www.nginx.com/blog/using-nginx-logging-for-application-performance-monitoring/#var_upstream_response_time

**Comment by Okiharaherbst** [comment]
I was hoping for a switch that would not involve introducing Nginx or Apache.

If the request user agent is in log output, response time would like a reasonable log output datum as well.

**Reply by wolfgangwalther** [reply]
Currently we use the [Common Log Format](https://en.wikipedia.org/wiki/Common_Log_Format) - extended with referer and user-agent:

> The format is extended by the Combined Log Format with [referer](https://en.wikipedia.org/wiki/HTTP_referer) and [user-agent](https://en.wikipedia.org/wiki/User-agent) fields.

This format does not contain response time.
